scipy/optimize/tests/test_hessian_update_strategy.py

```diff
@@ -91,14 +91,19 @@ def test_hessian_initialization(self):
     # Hessian update. Hence no update is
     # skiped or damped.
 
-    def test_initialize_catch_complex(self):
+    def test_initialize_catch_illegal(self):
         ndims = 3
-        init_scales = (complex(3.14),
+        init_scales = (complex(3.14),  # no complex allowed
                        np.array([3.2, 2.3, 1.2]).astype(np.complex128),
                        np.array([[43, 24, 33],
                                  [24, 36, 44, ],
-                                 [33, 44, 37, ]]).astype(np.complex128)
-                       )
+                                 [33, 44, 37, ]]).astype(np.complex128),
+                       np.array([[43, 24, 33]]),  # not square
+                       np.array([[43, 24, 33],
+                                 [24.1, 36, 44, ],
+                                 [33, 44, 37, ]]),  # not symmetric
```

Comment:
Sorry for being not verbose enough. At least in `scipy.stats`, the preferred way to test such input validation errors is to use `pytest.raises(ErroType, match="error message")`. See for example [here](https://github.com/scipy/scipy/blob/63ca1a1ccea554cedd0687e986a12cc6d27c924a/scipy/stats/tests/test_multivariate.py#L857) for a similar case. Could you please adapt your tests to this pattern? Then I will approve from the `optimize` side.

Reply:
Sure, I prefer that too. I've parametrized in a loop, but I can also use the pytest.mark.parametrized if preferred